x-pack/plugins/elastic_assistant/server/lib/langchain/executors/types.ts

```diff
@@ -14,14 +14,21 @@ import type { LangChainTracer } from 'langchain/callbacks';
 import { RequestBody, ResponseBody } from '../types';
 
 export interface AgentExecutorParams {
+  alertsIndexPattern?: string;
   actions: ActionsPluginStart;
+  allow?: string[];
+  allowReplacement?: string[];
+  assistantLangChain: boolean;
```

Comment:
nit: kinda weird this has to get injected since we're already in 'langchain' land, but makes sense as required by `getApplicableTools`. This should get cleaner as we rework tool registration, feature flags, and remove the non-langchain path.